python-backend/app/api/mines/variances/resources/variance_document_upload.py

```diff
@@ -0,0 +1,96 @@
+    # Methods don't share any common parser args, so none are defined here
+    parser = CustomReqparser()
+
+    @api.doc(description='Request a document_manager_guid for uploading a document')
+    @requires_any_of([MINE_CREATE, MINESPACE_PROPONENT])
+    def post(self, mine_guid, variance_id):
+        # DocumentManager requires parser data, but no arguments are required
+        data = self.parser.parse_args()
+        metadata = self._parse_request_metadata()
+        if not metadata or not metadata.get('filename'):
+            raise BadRequest('Filename not found in request metadata header')
+
+        # Save file
+        mine = Mine.find_by_mine_guid(mine_guid)
+        document_name = metadata.get('filename')
+        data = {
+            'folder': f'mines/{mine.mine_guid}/variances',
+            'pretty_folder': f'mines/{mine.mine_no}/variances',
+            'filename': document_name
+        }
+        document_manager_URL = f'{current_app.config["DOCUMENT_MANAGER_URL"]}/document-manager'
+
+        resp = requests.post(
+            url=document_manager_URL,
+            headers={key: value
+                     for (key, value) in request.headers if key != 'Host'},
+            data=data,
+            cookies=request.cookies,
+        )
+
+        response = Response(str(resp.content), resp.status_code, resp.raw.headers.items())
+        return response
+
+
+    @api.doc(
+        description='Associate an uploaded file with a variance.',
+        params={
+            'mine_guid': 'guid for the mine with which the variance is associated',
+            'variance_id': 'ID for the variance to which the document should be associated'
+        })
+    @api.marshal_with(VARIANCE_MODEL, code=200)
+    @requires_any_of([MINE_CREATE, MINESPACE_PROPONENT])
+    def put(self, mine_guid, variance_id):
+        # Arguments required by MineDocument
+        self.parser.add_argument('document_name', type=str, required=True)
+        self.parser.add_argument('document_manager_guid', type=str, required=True)
+
+        variance = Variance.find_by_variance_id(variance_id)
+
+        data = self.parser.parse_args()
+        document_name = data.get('document_name')
+        document_manager_guid = data.get('document_manager_guid')
+
+        # Register new file upload
+        mine_doc = MineDocument(
+            mine_guid=mine_guid,
+            document_manager_guid=document_manager_guid,
+            document_name=document_name)
+
+        if not mine_doc:
+            raise BadRequest('Unable to register uploaded file as document')
+
+        variance.documents.append(mine_doc)
+        variance.save()
+        return variance
+
+    def _parse_request_metadata(self):
+        request_metadata = request.headers.get("Upload-Metadata")
+        metadata = {}
+        if not request_metadata:
+            return metadata
+
+        for key_value in request_metadata.split(","):
```

Comment:
`key_value` looks odd, is this equal to `key, value`?

Reply:
I don't believe so. Based on the next line, it looks like it's a string in the form of "{key} {value}". I didn't look at the exact implementation logic here. It was just copied from `python-backend/app/api/permits/permit_amendment/resources/permit_amendment_document.py` and then worked as-expected.

Reply:
oh, i looked the next reference and missed it, that make sense. thanks.